src/Illuminate/Database/DatabaseTransactionsManager.php

```diff
@@ -89,36 +100,14 @@ public function commit($connection)
      */
     public function addCallback($callback)
     {
-        if ($current = $this->callbackApplicableTransactions()->last()) {
-            return $current->addCallback($callback);
+        // If there are no transactions, we'll run the callbacks right away. Also, we'll run it
+        // right away when we're in test mode and we only have the wrapping transaction. For
+        // every other case, we'll queue up the callback to run after the commit happens.
+        if ($this->transactions->isEmpty() || ($this->afterCommitCallbacksRunningInTestTransaction && $this->transactions->count() == 1)) {
+            return $callback();
         }
 
-        $callback();
-    }
-
-    /**
-     * Specify that callbacks should ignore the given transaction when determining if they should be executed.
-     *
-     * @param  \Illuminate\Database\DatabaseTransactionRecord  $transaction
-     * @return $this
-     */
-    public function callbacksShouldIgnore(DatabaseTransactionRecord $transaction)
```

Comment:
Should I add back this method signature?

Reply:
I've added this method back and we're marking it as deprecated. It forwards to the new `withAfterCommitCallbacksInTestTransactionAwareMode` method for now.